classes/assessment/lti.php

```diff
@@ -0,0 +1,96 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace local_sitsgradepush\assessment;
+
+/**
+ * Class for LTI assessment.
+ *
+ * @package    local_sitsgradepush
+ * @copyright  2024 onwards University College London {@link https://www.ucl.ac.uk/}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author     David Watson <david-watson@ucl.ac.uk>
+ */
+class lti extends activity {
+
+    /**
+     * Get all participants.
+     *
+     * @return array
+     */
+    public function get_all_participants(): array {
+        if (!self::check_assessment_validity()->valid) {
+            return [];
+        }
+        $context = \context_module::instance($this->coursemodule->id);
+        return get_enrolled_users($context, 'mod/lti:view');
```

Comment:
don't teachers also have the view permission?

Reply:
They do.  I added commit 366184b to filter teacher out of this method for LTI, Lesson and H5P plugin.  This gets around the fact that we don't have a student only capability like 'mod/xxx:submit' in those activity types